add new blog post about BELLATRIX test automation framework (#1153)
SeleniumHQ/seleniumhq.github.io: website_and_docs/content/blog/2022/bellatrix-test-automation-framework/index.md

---
title: "BELLATRIX Test Automation Framework for C# and JAVA"
linkTitle: "BELLATRIX Test Automation Framework for C# and JAVA"

> date: 2022-11-10

Reviewer: ```suggestion
date: 2022-11-16
```

tags: ["webdriver", "framework", "ecosystem", "java", "dotnet"]
categories: ["technical", "general"]
author: Anton Angelov ([@angelovstanton](https://www.linkedin.com/in/angelovstanton/))
description: >

>   Customize and extend our cross-platform .NET 6 and JAVA test automation framework to perfectly fit your needs.
>   Start on top of hundreds of best practice features and integrations.

Reviewer: ```suggestion
  Customize and extend BELLATRIX, a cross-platform .NET 6 and JAVA test automation framework to perfectly fit your needs.
  Start on top of hundreds of best practices features and integrations.
```

---

> title: "Selenium Ecosystem - BELLATRIX Test Automation Framework for C# and JAVA"

Reviewer: ```suggestion
title: "BELLATRIX Test Automation Framework for C# and JAVA"
```

> linkTitle: "Selenium Ecosystem - BELLATRIX Test Automation Framework for C# and JAVA"

Reviewer: ```suggestion
linkTitle: "BELLATRIX Test Automation Framework for C# and JAVA"
```

> date: 2022-09-01

Reviewer: ```suggestion
date: 2022-11-10
```

tags: ["webdriver", "framework", "ecosystem", "java", "dotnet"]
categories: ["technical", "general"]
author: Anton Angelov ([@angelovstanton](https://www.linkedin.com/in/angelovstanton/))
description: >
  Customize and extend our cross-platform .NET 6 and JAVA test automation framework to perfectly fit your needs.
  Start on top of hundreds of best practice features and integrations.
---
Over the last decade, a large ecosystem of Open Source projects has sprouted up around Selenium. Selenium is often used for automating web applications for testing purposes, but it does not include a testing framework.
Nowadays, Selenium Ecosystem initiatives try to give popularity to popular open-source test automation frameworks maintained by people outside of core Selenium maintainers.

> One of these frameworks is BELLATRIX, developed and maintained by [Automate The Planet](https://bellatrix.solutions/about/) and invented by [Anton Angelov](https://www.linkedin.com/in/angelovstanton/). It has two versions - C# and Java.

Reviewer: ```suggestion
One of these frameworks is BELLATRIX, invented by [Anton Angelov](https://www.linkedin.com/in/angelovstanton/). It has two versions - C# and Java.
```

Reviewer: I just want to avoid people commenting that we have a link to a company and thinking that we are promoting the company, whereas we want to promote the Open Source framework.

A framework can be based on several libraries implementing several APIs, but unlike the normal use of an API, the access to the behavior built into the framework is mediated by extending its content with new classes plugged into the framework itself. Moreover, the overall program flow of control can be out of the control of the caller and in the hands of the framework by inversion of control or a similar mechanism.

#### Definition Test Automation Framework ###
Various guidelines, coding standards, concepts, processes, practices, project hierarchies, modularity, etc. to support automated testing. The user can follow these guidelines while automating application to take advantages of various productive results.

> 

Reviewer: While I can say that agree with this content, it is not the purpose of the blog to define a test framework. I'd prefer if we go more to the point and highlight BELLATRIX without loosing time in definitions.

If you wish, you can leave something along the lines of "A testing framework is an abstraction in which common code providing generic functionality (which can be selectively overridden) for testing different aspects of our applications- UI, API, security, performance and many other." 

And then connecting that with BELLATRIX.

### BELLATRIX Test Automation Framework ###
The first version of **[BELLATRIX](https://bellatrix.solutions/)** appeared on 26 December 2017. It was available only for C# initially but written on new back then .NET Core, allowing the framework to be used on all major operating systems (cross-platform).
One huge advantage of BELLATRIX is cross-technology readiness. It allows you to write tests for different technologies such as Web, Mobile, Desktop, and API. For me, this also includes a similar API. In BELLATRIX, we strive the API for different modules to be as identical as possible. Here by API, I don't mean web API.

### Definition API ###

> API or application programming interface is a set of functions and procedures that allow the creation of applications which access the features or data of an operating system, application, or other service. A good API makes it easier to develop a computer program by providing all the building blocks, which are then put together by the programmer.

Reviewer: I believe this could be summarized and added to the previous paragraph.

> ###Multiple Test Environments Configuration###

Reviewer: From this point on, it is hard for me to grasp why BELLATRIX is a good alternative. Wouldn't be better to show a simple "how to get started" so the reader gets interested about it, and after that mention the different features in a bullet list with a short description (like dynamic tests cases, email testing, etc...). 

When going too much into details, one misses the point of how to get started with the framework and what else could it offer me. I hope this makes sense to you.
